lenses/authselectpam.aug

```diff
@@ -0,0 +1,106 @@
+(*
+Module: AuthselectPam
+  Parses /etc/authselect/custom/*/*-auth and
+  /etc/authselect/custom/*/postlogin files
+
+Author: Heston Snodgrass <heston.snodgrass@puppet.com> based on pam.aug by David Lutterkort <lutter@redhat.com>
+
+About: Reference
+  This lens tries to keep as close as possible to `man pam.conf` where
+  possible. This lens supports authselect templating syntax as
+  can be found in `man authselect-profiles`.
+
+About: Licence
+  This file is licensed under the LGPL v2+, like the rest of Augeas.
+
+About: Lens Usage
+
+About: Configuration files
+  This lens also autoloads /etc/authselect/custom/*/*-auth and
+  /etc/authselect/custom/*/postlogin because these files are PAM template
+  files on machines that have authselect custom profiles.
+*)
+module AuthselectPam =
+  autoload xfm
+
+  let eol = Util.eol
+  let indent = Util.indent
+  let space = del /([ \t]|\\\\\n)+/ " "
+  let reg_ws = del /([ \t])/ " "
+
+  (* For the control syntax of [key=value ..] we could split the key value *)
+  (* pairs into an array and generate a subtree control/N/KEY = VALUE      *)
+  (* The valid control values if the [...] syntax is not used, is          *)
+  (*   required|requisite|optional|sufficient|include|substack             *)
+  (* We allow more than that because this list is not case sensitive and   *)
+  (* to be more lenient with typos                                         *)
+  let control = /(\[[^]{}#\n]*\]|[a-zA-Z]+)/
+  let word = /([^{}# \t\n\\]|\\\\.)+/
+  (* Allowed types *)
+  let types = /(auth|session|account|password)/i
+
+  (* This isn't entirely right: arguments enclosed in [ .. ] can contain  *)
+  (* a ']' if escaped with a '\' and can be on multiple lines ('\')       *)
+  let argument = /(\[[^]{}#\n]+\]|[^[{#\n \t\\][^#\n \t\\]*)/
```

Comment:
From what I can tell, this is all copied from `pam.aug`. I would suggest to remove this and refer to these variables directly from the `Pam` module, see below.

Reply:
Thanks for the feedback, I've incorporated it into the last commit.